ocaml/tests/test_vm_helpers.ml

```diff
@@ -573,6 +599,352 @@ let test_get_group_key_vgpu_and_netsriov () =
           Alcotest.fail "Test-failure: Unexpected Group Key in test"
   )
 
+let test_get_group_key_anti_affinity_and_vgpu_and_netsriov () =
+  on_pool_of_intel_i350 (fun __context _ _ _ ->
+      let group = List.hd (Db.GPU_group.get_all ~__context) in
+      let vm = make_vm_with_vgpu_in_group ~__context VGPU_T.k100 group in
+      let sriov_network =
+        List.find
+          (fun network ->
+            Xapi_network_sriov_helpers.is_sriov_network ~__context ~self:network
+          )
+          (Db.Network.get_all ~__context)
+      in
+      let (_ : API.ref_VIF) =
+        T.make_vif ~__context ~vM:vm ~network:sriov_network ()
+      in
+      let anti_affinity_group =
+        T.make_vm_group ~__context ~placement:`anti_affinity ()
+      in
+      Db.VM.set_groups ~__context ~self:vm ~value:[anti_affinity_group] ;
+      match Xapi_vm_helpers.get_group_key ~__context ~vm with
+      | `AntiAffinity _ ->
+          ()
+      | _ ->
+          Alcotest.fail "Test-failure: Unexpected Group Key in test"
+  )
+
+type vm_info = {name: string; host: string; group: string; state: string}
```

Comment:
Can we put related code below for `vm_anti_affinity_ranked_grp_tests` into one module, I think it will make it more clear, here is a simple example: [vm_anti_affinity_ranked_grp_tests](https://github.com/xapi-project/xen-api/blob/master/ocaml/tests/test_pool_periodic_update_sync.ml#L24)